---
title: "seacr_peakReprod.Rmd"
output: output=github_document
date: "2024-08-23"
---

```{setting up project directory}
projPath = ("/rds/general/user/mml120/ephemeral/p2/CR_000")
```

```{Packages}
library(ggpubr)
library(ggplot2)
library(dplyr)
library(viridis)
library(GenomicRanges)
library(chromVAR)
```

```{Sample List}
sampleList = c("CR_000_C2A","CR_000_C2B", "CR_000_C5A", "CR_000_C5B", "CR_000_E2A", "CR_000_E2B", "CR_000_E5A", "CR_000_E5B", "CR_000_I2A", "CR_000_I2B", "CR_000_I5A", "CR_000_I5B", "CR_000_M2A", "CR_000_M2B", "CR_000_M5A", "CR_000_M5B")

histList = "CR_000"
```

```{Number of Peaks}
peakN = c()
peakWidth = c()
peakType = c("control", "top0.01")
for(hist in sampleList){
  histInfo = strsplit(hist, "_")[[1]]
  replicate = substr(histInfo[3], nchar(histInfo[3]), nchar(histInfo[3]))
  sampleName = paste0(histInfo[1], "_", histInfo[2], "_", substr(histInfo[3], 1, nchar(histInfo[3])-1))
  if(!(histInfo[3] %in% c("I2A", "I2B", "I5A", "I5B"))){
    for(type in peakType){
      peakInfo = read.table(paste0(projPath, "/peakCalling/SEACR_peaks/", hist, "_seacr_", type, ".peaks.stringent.bed"), header = FALSE, fill = TRUE)  %>% mutate(width = abs(V3-V2))
      peakN = data.frame(peakN = nrow(peakInfo), peakType = type, Sample = sampleName, Replicate = replicate) %>% rbind(peakN, .)
      peakWidth = data.frame(width = peakInfo$width, peakType = type, Sample = sampleName, Replicate = replicate)  %>% rbind(peakWidth, .)
    }
  }
}
peakN %>% dplyr::select(Sample, Replicate, peakType, peakN)
```

```{Peak Overlap}
histL = c("CR_000_C2", "CR_000_C5", "CR_000_E2", "CR_000_E5", "CR_000_M2", "CR_000_M5")
peakType = c("control", "top0.01")
repL = paste0(c("A","B"))

peakOverlap = c()
for(type in peakType){
  for(hist in histL){
    overlap.gr = GRanges() 
    histInfo = strsplit(hist, "_")[[1]]
    sampleName = paste0(histInfo[1], "_", histInfo[2], "_", substr(histInfo[3], 1, nchar(histInfo[3])-1))
    for(rep in repL){
      peakInfo = read.table(paste0(projPath, "/peakCalling/SEACR_peaks/", hist, rep, "_seacr_", type, ".peaks.stringent.bed"), header = FALSE, fill = TRUE)
      peakInfo.gr = GRanges(peakInfo$V1, IRanges(start = peakInfo$V2, end = peakInfo$V3), strand = "*")
      if(length(overlap.gr) >0){
        overlap.gr = overlap.gr[findOverlaps(overlap.gr, peakInfo.gr)@from]
      }else{
        overlap.gr = peakInfo.gr
        
      }
    }
    peakOverlap = data.frame(peakReprod = length(overlap.gr), Sample = sampleName, peakType = type) %>% rbind(peakOverlap, .)
  }
}

peakReprod = left_join(peakN, peakOverlap, by = c("Sample", "peakType")) %>% mutate(peakReprodRate = peakReprod/peakN * 100)
peakReprod %>% dplyr::select(Sample, Replicate, peakType, peakN, peakReprodNum = peakReprod, peakReprodRate)

write.csv(peakReprod, paste0("/rds/general/user/mml120/ephemeral/p2/CR_000/csvTable/scaleFactorSummary.csv", sep=""), row.names = FALSE)
```

```{FRiP Scores}
bamDir = paste0(projPath, "/alignment/bam")
inPeakData = c()
## overlap with bam file to get count
for(hist in histL){
  histInfo = strsplit(hist, "_")[[1]]
  sampleName = paste0(histInfo[1], "_", histInfo[2], "_", substr(histInfo[3], 1, nchar(histInfo[3])-1))    
  for(rep in repL){
    peakRes = read.table(paste0(projPath, "/peakCalling/SEACR_peaks/", sampleName, rep, "_seacr_top0.01.peaks.stringent.bed"), header = FALSE, fill = TRUE)
    peak.gr = GRanges(seqnames = peakRes$V1, IRanges(start = peakRes$V2, end = peakRes$V3), strand = "*")
    bamFile = paste0(bamDir, "/", sampleName, rep, "_bowtie2.mapped.bam")
    fragment_counts <- getCounts(bamFile, peak.gr, paired = TRUE, by_rg = FALSE, format = "bam")
    inPeakN = counts(fragment_counts)[,1] %>% sum
    inPeakData = rbind(inPeakData, data.frame(inPeakN = inPeakN, Sample = sampleName, Replicate = rep))
  }
}

frip = left_join(inPeakData, alignResult, by = c("Sample", "Replicate")) %>% mutate(frip = inPeakN/MappedFragNum_T2T * 100)
frip %>% dplyr::select(Sample, Replicate, SequencingDepth, MappedFragNum_T2T, AlignmentRate_T2T, FragInPeakNum = inPeakN, FRiPs = frip)
```

```{Reproducibility Figure}
fig7A = peakN %>% ggplot(aes(x = Sample, y = peakN, fill = Sample)) +
  geom_boxplot() +
  geom_jitter(aes(color = Replicate), position = position_jitter(0.15)) +
  facet_grid(~peakType) +
  scale_fill_viridis(discrete = TRUE, begin = 0.1, end = 0.55, option = "magma", alpha = 0.8) +
  scale_color_viridis(discrete = TRUE, begin = 0.1, end = 0.9) +
  theme_bw(base_size = 18) +
  ylab("Number of Peaks") +
  xlab("")

fig7B = peakWidth %>% ggplot(aes(x = Sample, y = width, fill = Sample)) +
  geom_violin() +
  facet_grid(Replicate~peakType) +
  scale_fill_viridis(discrete = TRUE, begin = 0.1, end = 0.55, option = "magma", alpha = 0.8) +
  scale_color_viridis(discrete = TRUE, begin = 0.1, end = 0.9) +
  scale_y_continuous(trans = "log", breaks = c(400, 3000, 22000)) +
  theme_bw(base_size = 18) +
  ylab("Width of Peaks") +
  xlab("")

fig7C = peakReprod %>% ggplot(aes(x = Sample, y = peakReprodRate, fill = Sample, label = round(peakReprodRate, 2))) +
  geom_bar(stat = "identity") +
  geom_text(vjust = 0.1) +
  facet_grid(Replicate~peakType) +
  scale_fill_viridis(discrete = TRUE, begin = 0.1, end = 0.55, option = "magma", alpha = 0.8) +
  scale_color_viridis(discrete = TRUE, begin = 0.1, end = 0.9) +
  theme_bw(base_size = 18) +
  ylab("% of Peaks Reproduced") +
  xlab("")

fig7D = frip %>% ggplot(aes(x = Sample, y = frip, fill = Sample, label = round(frip, 2))) +
  geom_boxplot() +
  geom_jitter(aes(color = Replicate), position = position_jitter(0.15)) +
  scale_fill_viridis(discrete = TRUE, begin = 0.1, end = 0.55, option = "magma", alpha = 0.8) +
  scale_color_viridis(discrete = TRUE, begin = 0.1, end = 0.9) +
  theme_bw(base_size = 18) +
  ylab("% of Fragments in Peaks") +
  xlab("")

peakCallingFig <- ggarrange(fig7A, fig7B, fig7C, fig7D, ncol = 2, nrow=2, common.legend = TRUE, legend="bottom")
ggsave(filename = paste0("peakCallingFig.pdf"), plot = peakCallingFig, device = "pdf", width = 100, height = 50, units = "cm", dpi = 300)
```
